de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/console/ShareCommand.java

```diff
@@ -0,0 +1,54 @@
+package de.fu_berlin.inf.dpp.server.console;
+
+import de.fu_berlin.inf.dpp.filesystem.IWorkspace;
+import de.fu_berlin.inf.dpp.server.filesystem.ServerProjectImpl;
+import de.fu_berlin.inf.dpp.session.ISarosSession;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.session.User;
+import java.io.PrintStream;
+import java.util.List;
+import org.apache.log4j.Logger;
+
+public class ShareCommand extends ConsoleCommand {
+  private static final Logger log = Logger.getLogger(ShareCommand.class);
+  private final ISarosSessionManager sessionManager;
+  private final IWorkspace workspace;
+
+  public ShareCommand(
+      ISarosSessionManager sessionManager, IWorkspace workspace, ServerConsole console) {
+    this.sessionManager = sessionManager;
+    this.workspace = workspace;
+    console.registerCommand(this);
+  }
+
+  @Override
+  public String identifier() {
+    return "share";
+  }
+
+  @Override
+  public String help() {
+    return "share <PATH>... - Share projects relative to the workspace with session participants";
+  }
+
+  @Override
+  public void execute(List<String> args, PrintStream out) {
+    ISarosSession session = sessionManager.getSession();
+
+    if (session == null) {
+      log.error("No Session running, cannot add any resources");
+      return;
+    }
+
+    try {
+      args.stream()
```

Comment:
This is really a good use case to ask the question if we should use lambdas at all.

1. On line 47 you use still the normal iterator for:each, why no lambda ?

2. Is it really your intention to do

for (arg : args) pl.add(new XYZ(arg))
for (p : pl) someFunction(pl)

or

for (arg : args) someFunction(new XYZ(arg))

I am just asking because lets assume you want that an issue with an argument will not affect other arguments so you would normally write:

for (arg : args) try {someFunction(new XYZ(arg))}catch(...){}

but am I right that you have to write 2 catch blocks when using lambdas ?

so arg.stream().map(x -> try{...}catch(...){}).forEach(x -> try{...}catch(...){}))

Reply:
Instead of making this issue about the usage of lambdas, you could have just pointed out, that I am missing separate error handling for every argument (similar to #282, where I was not validating JIDs), because yes, it is indeed desirable that a malformed argument is either blocking the whole command or just gets skipped instead of aborting the rest.

And indeed error handling with lambdas is quite difficult (partly due to java's exception system), because these iterators are lazy, so all previous arguments will have been processed before the exception is thrown and aborts the whole stream. Also a try-catch-block inside a lambda would not help you at all, because your lambda would still need to return a `ServerProjectImpl` even if you catch the exception.

So yes, lambdas are not a good solution to this problem, but they would have been just fine, if that particular mistake would not have been the case. Just like they did work out in the final (and merged) variant of #282.

So instead of arguing over code, that is merely a matter of personal preference, I would kindly ask you to point out the actual mistake of mine next time.

Reply:
I am sorry but that was really a discussion about lambdas and not about error handling in non production code.

try { doStuff() } catch(Exception weDoNotCare){} just continue is a common pattern for test code.

As this is debug stuff I let the author decide how much error handling is done.